File: app/src/main/java/com/example/study/BaseActivity.java
package com.example.study;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;

import androidx.annotation.Nullable;

import com.example.study.base.activity.BaseFileActivity;
import com.example.study.base.activity.BasePermissionActivity;
import com.example.study.util.LogUtil;

/**
 * Copyright , 2015-2019 <br>
 * Author: 陈刘磊 1070379530@qq.com <br>
 * Date: 2019/6/18 14:35    <br>
 * Description: BaseActivity  <br>
 */
public class BaseActivity extends BaseFileActivity {
    protected String TAG = getClass().getSimpleName();


    public static void start(Context context, Class<? extends BaseActivity> c) {
        Intent starter = new Intent(context, c);
        context.startActivity(starter);
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        LogUtil.Companion.i("life", TAG + "-----onCreate");
        super.onCreate(savedInstanceState);
    }

    @Override
    protected void onNewIntent(Intent intent) {
        super.onNewIntent(intent);
        LogUtil.Companion.i("life", TAG + "-----onNewIntent");
    }

    @Override
    protected void onRestart() {
        super.onRestart();
        LogUtil.Companion.i("life", TAG + "-----onRestart");
    }

    @Override
    protected void onStart() {
        super.onStart();
        LogUtil.Companion.i("life", TAG + "-----onStart");

    }

    @Override
    protected void onResume() {
        super.onResume();
        LogUtil.Companion.i("life", TAG + "-----onResume");

    }

    @Override
    protected void onPause() {
        super.onPause();
        LogUtil.Companion.i("life", TAG + "-----onPause");
    }

    @Override
    protected void onStop() {
        super.onStop();
        LogUtil.Companion.i("life", TAG + "-----onStop");
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        LogUtil.Companion.i("life", TAG + "-----onDestroy");

    }

}

File: cameralib/src/main/java/com/example/cameralib/base/activity/BaseActivity.kt
package com.example.cameralib.base.activity

import android.content.Context
import android.content.Intent
import android.os.Bundle
import com.example.cameralib.util.LogUtil

/**
 * Copyright , 2015-2019 <br></br>
 * Author: 陈刘磊 1070379530@qq.com <br></br>
 * Date: 2019/6/18 14:35    <br></br>
 * Description: BaseActivity  <br></br>
 */
open class BaseActivity : BaseFileActivity() {
    protected var TAG = javaClass.simpleName


    companion object {
        fun start(context: Context, c: Class<out BaseActivity>) {
            val starter = Intent(context, c)
            context.startActivity(starter)
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        LogUtil.i("life", "$TAG-----onCreate")
        super.onCreate(savedInstanceState)
    }

    override fun onNewIntent(intent: Intent) {
        super.onNewIntent(intent)
        LogUtil.i("life", "$TAG-----onNewIntent")
    }

    override fun onRestart() {
        super.onRestart()
        LogUtil.i("life", "$TAG-----onRestart")
    }

    override fun onStart() {
        super.onStart()
        LogUtil.i("life", "$TAG-----onStart")

    }

    override fun onResume() {
        super.onResume()
        LogUtil.i("life", "$TAG-----onResume")

    }

    override fun onPause() {
        super.onPause()
        LogUtil.i("life", "$TAG-----onPause")
    }

    override fun onStop() {
        super.onStop()
        LogUtil.i("life", "$TAG-----onStop")
    }

    override fun onDestroy() {
        super.onDestroy()
        LogUtil.i("life", "$TAG-----onDestroy")

    }



}
